landoapi/models/landing_job.py

```diff
@@ -144,20 +126,52 @@ class LandingJob(Base):
     #    ["", ""]
     formatted_replacements = db.Column(JSONB, nullable=True)
 
+    revisions = db.relationship(
+        "Revision",
+        secondary=revision_landing_job,
+        back_populates="landing_jobs",
+        order_by="revision_landing_job.columns.index",
+    )
+
     @property
     def landing_path(self) -> list[tuple[int, int]]:
-        return [(int(r), self.revision_to_diff_id[r]) for r in self.revision_order]
+        return [(revision.revision_id, revision.diff_id) for revision in self.revisions]
+
+    @property
+    def landing_path_compat(self):
```

Comment:
This method and `landing_path` do different things. `landing_path_compat` is called in `serialize` to replace previous functionality that was coded in that method itself (to set the `landing_path` value, which itself is different from `self.landing_path`). This is probably a poor naming choice, however `serialized_landing_path` is a better name here so I can change it.

An easier way to picture this is if we had moved the original functionality from `self.serialize` to its own method, and then made it legacy-compatible. See deleted lines 277-280 in `landoapi/models/landing_job.py`. I.e., this is already doing both the legacy and non-legacy functionality, and does not have much to do with the original `self.landing_path` method (other than making use of it, instead of duplicating that same functionality in `self.serialize` as in the previous implementation.)